src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -28,18 +29,33 @@ public CollectionTallyResult(List<object> missingItems, List<object> extraItems)
 
         private readonly NUnitEqualityComparer comparer;
 
+        private bool _sorted = false;
+
         /// <summary>The result of the comparison between the two collections.</summary>
         public CollectionTallyResult Result
         {
             get
             {
-                return new CollectionTallyResult(
-                    new List<object>(_missingItems),
-                    new List<object>(_extraItems));
+                var missingItems = new List<object>(_missingItems.Count);
+                foreach (var o in _missingItems)
+                    missingItems.Add(o);
+
+                List<object> extraItems = new List<object>();
+                if(_sorted)
+                {
+                    for (int index = _extraItems.Count - 1; index >= 0; index--)
```

Comment:
All of the extra backwards iterations of collections is extra complexity but the only way I could not revert the performance gains @mikkelbu made fixing #2598